docs/usage/examples/opentelemetry.md

```diff
@@ -0,0 +1,278 @@
+# OpenTelemetry
+
+Requirements:
+
+- docker-compose
+
+## Prepare setup
+
+Create a `docker-compose.yaml` and `otel-collector-config.yml` file as seen below in a folder.
+
+`docker-compose.yaml`:
+
+```yaml
+version: '3'
+services:
+  # Jaeger
+  jaeger:
+    image: jaegertracing/all-in-one:latest
+    ports:
+      - '16686:16686'
+      - '14250'
+
+  otel-collector:
+    image: otel/opentelemetry-collector-contrib:0.52.0
+    command: ['--config=/etc/otel-collector-config.yml']
+    volumes:
+      - ./otel-collector-config.yml:/etc/otel-collector-config.yml
+    ports:
+      - '1888:1888' # pprof extension
+      - '13133:13133' # health_check extension
+      - '55679:55679' # zpages extension
+      - '4318:4318' # OTLP HTTP
+      - '4317:4317' # OTLP GRPC
+      - '9123:9123' # Prometheus exporter
+    depends_on:
+      - jaeger
+```
+
+`otel-collector-config.yml`:
+
+```yaml
+receivers:
+  otlp:
+    protocols:
+      grpc:
+      http:
+
+exporters:
+  jaeger:
+    endpoint: jaeger:14250
+    tls:
+      insecure: true
+  logging:
+  prometheus:
+    endpoint: '0.0.0.0:9123'
+
+processors:
+  batch:
+  spanmetrics:
+    metrics_exporter: prometheus
+    latency_histogram_buckets: [10ms, 100ms, 250ms, 1s, 30s, 1m, 5m]
+    dimensions:
+      - name: http.method
+      - name: http.status_code
+      - name: http.host
+    dimensions_cache_size: 1000
+    aggregation_temporality: 'AGGREGATION_TEMPORALITY_CUMULATIVE'
+
+extensions:
+  health_check:
+  pprof:
+  zpages:
+
+service:
+  extensions: [pprof, zpages, health_check]
+  pipelines:
+    traces:
+      receivers: [otlp]
+      exporters: [jaeger, logging]
+      processors: [spanmetrics, batch]
+
+    metrics:
+      receivers: [otlp]
+      exporters: [prometheus]
+```
+
+Start setup using this command inside the folder containing the files created in the earlier steps:
+
+```
+docker-compose up
+```
+
+This command will start an [OpenTelemetry Collector](https://github.com/open-telemetry/opentelemetry-collector-contrib) and an instance of [Jaeger](https://www.jaegertracing.io/).
+
+Jaeger will be now reachable under [http://localhost:16686](http://localhost:16686).
+
+## Run Renovate with OpenTelemetry
+
+To start Renovate with OpenTelemetry enabled run following command, after pointing to your `config.js` config file:
+
+```
+docker run \
+  --rm \
+  -e OTEL_EXPORTER_OTLP_ENDPOINT=http://localhost:4318 \
+  -v "/path/to/your/config.js:/usr/src/app/config.js" \
+  renovate/renovate:latest
+```
+
+You should now see `trace_id` and `span_id` fields in the logs.
```

Comment:
Will it be for all messages, or only ones you've explicitly altered in source?

Reply:
All lines written by `bunyan` will be modified.